iroh/src/rpc_protocol.rs

```diff
@@ -155,13 +155,18 @@ pub struct BlobListRequest;
 
 /// A response to a list blobs request
 #[derive(Debug, Serialize, Deserialize)]
-pub struct BlobListResponse {
-    /// Location of the blob
-    pub path: String,
-    /// The hash of the blob
-    pub hash: Hash,
-    /// The size of the blob
-    pub size: u64,
+pub enum BlobListResponse {
+    /// A blob
+    Item {
+        /// Location of the blob
+        path: String,
+        /// The hash of the blob
+        hash: Hash,
+        /// The size of the blob
+        size: u64,
+    },
+    /// An error occurred while listing blobs
+    IoError(RpcError),
```

Comment:
why is this not `Result<BlobListResponse, RpcError>` this new enum makes the (public facing) api quite confusing